lib/streamlit/caching.py

```diff
@@ -659,17 +680,25 @@ def has_changes(self):
                 self._allow_output_mutation,
                 code,
                 [caller_lineno + 1, caller_lineno + len(lines)],
+                hash_funcs,
             )
             self.update(value)
         except (CacheKeyNotFoundError, CachedObjectWasMutatedError):
             if self._allow_output_mutation and not self._persist:
                 # If we don't hash the results, we don't need to use exec and just return True.
                 # This way line numbers will be correct.
-                _write_to_cache(key, self, False, True, None)
+                _write_to_cache(key, self, False, True, None, None)
```

Comment:
I know this is not your doing, but it would be nice to use kwargs for each of these arguments. Otherwise it's impossible to tell exactly what this line is doing...